### PR TITLE
`file()`, `stdout()`: fix log sources getting stuck

### PR DESCRIPTION
**Pull request by MrAnno**
If a partial write occurs, the written messages must be acknowledged.

It is important to note that a message may be written partially.
In that case, the message will be acknowledged once the complete write is successful. If write errors occur during this process, the partially written message may be duplicated.

### CODE REVIEW

File: modules/affile/logproto-file-writer.c [hunk 1]
```diff
@@ -45,6 +45,48 @@ typedef struct _LogProtoFileWriter
...
+    {
+      self->partial_pos += rc;
+      *status = LPS_PARTIAL;
+      return FALSE;
+    }
+
+  log_proto_client_msg_ack(&self->super, self->partial_messages);
+  g_free(self->partial);
+  self->partial = NULL;
+  self->partial_messages = 0;
```
Comment (alltilla) [inline modules/affile/logproto-file-writer.c:86]: [optional]

As far as I can see, this was not part of the original code. I guess it is needed, but probably should be introduced in another commit.

**Review by alltilla — APPROVED** [review]
Awesome!

I had one small comment which you can ignore.

File: modules/affile/logproto-file-writer.c (hunk 1 above)
Comment (MrAnno) [reply modules/affile/logproto-file-writer.c:86]: It's unnecessary, it was just part of my thought process. I can either remove it or commit it separately.

Comment (alltilla) [reply modules/affile/logproto-file-writer.c:86]: I am okay with either, or keeping it like this, too. Let me know if I should merge it this way :)

Comment (MrAnno) [reply modules/affile/logproto-file-writer.c:86]: I've extracted it to a separate commit. Thanks.

**Review by alltilla — APPROVED** [review]

**Pull request merged**